app/data/data-chat-icon/src/main/kotlin/com/hedvig/android/data/chat/icon/GetChatIconAppStateUseCase.kt

```diff
@@ -0,0 +1,69 @@
+package com.hedvig.android.data.chat.icon
+
+import com.hedvig.android.data.chat.read.timestamp.ChatLastMessageReadRepository
+import com.hedvig.android.data.settings.datastore.SettingsDataStore
+import com.hedvig.android.featureflags.FeatureManager
+import com.hedvig.android.featureflags.flags.Feature
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.currentCoroutineContext
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.isActive
+
+interface GetChatIconAppStateUseCase {
+  fun invoke(): Flow<ChatIconAppState>
+}
+
+class GetChatIconAppStateUseCaseImpl(
+  private val settingsDataStore: SettingsDataStore,
+  private val featureManager: FeatureManager,
+  private val shouldShowChatIconUseCase: ShouldShowChatIconUseCase,
+  private val chatLastMessageReadRepository: ChatLastMessageReadRepository,
+) : GetChatIconAppStateUseCase {
+  override fun invoke(): Flow<ChatIconAppState> {
+    return shouldShowChatIconUseCase.invoke().flatMapLatest { showChat: Boolean ->
+      if (!showChat) {
+        flowOf(ChatIconAppState.Hidden)
+      } else {
+        combine(
+          featureManager.isFeatureEnabled(Feature.CHAT_BUBBLE),
```

Comment:
Hehe yeap you got me. It's not in unleash, it's just off for everyone in production. This was on purpose, but we could add it anytime.

Reply:
I'd say probably worth to add it in unleash and turn it off from there - so it would serve as a reminder at least that have this function and it would not get lost and forgotten in a year:)